### PR TITLE
Add variable fonts

### LINKED ISSUE #469: Variable font assets
Although our efforts to add Source Sans Variable to the current iteration have been somewhat frustrating (as documented in #452, which you should totally read before starting this). A major release gives us an opportunity to revise our strategy for this.

To keep us from stalling completely, I will be adding a separate issue for font loading strategies. This issue should prioritize getting it working in a simple way that sets us up for future success.

## Tasks

- [ ] Generate/procure/include necessary font files
  - [ ] Source Sans Variable (body copy)
    - [ ] WOFF2 / Regular
    - [ ] WOFF2 / Italic
  - [ ] Source Sans Pro (fallback where variable fonts aren't supported)
    - [ ] WOFF2
      - [ ] Regular
      - [ ] Semibold
      - [ ] Bold
      - [ ] Italic
    - [ ] WOFF
      - [ ] Regular
      - [ ] Semibold
      - [ ] Bold
      - [ ] Italic
  - [ ] Source Sans Code (code samples)
    - [ ] WOFF2 / Regular
    - [ ] WOFF / Regular
- [ ] Add CSS/Sass necessary for usage

### PR DESCRIPTION
**Pull request by calebeby**
Closes #469 

https://deploy-preview-506--cloudfour-patterns.netlify.com/?path=/story/demo-button--button-element

## Testing

- Apply different font weights and styles to the button using devtools to verify that the Source Sans Pro variable font is working correctly
- Change the button to use "Source Code Pro" and verify that it works (this is not a variable font)

Test in IE11 to see if the Source Sans Pro non-variable font fallbacks are loaded

### CODE REVIEW

**Comment by Paul-Hebert** [comment]
I know this is WIP but it's exciting!

File: src/base/_typography.scss [hunk 1]
```diff
@@ -14,4 +14,65 @@ html {
...
   text-size-adjust: none; /* 1 */
   word-wrap: break-word; /* 2 */
+  font-family: 'Source Sans Pro', sans-serif;
+
+  @supports (font-variation-settings: normal) {
+    font-family: 'Source Sans Pro Variable', sans-serif;
+  }
+}
+
+@font-face {
```
Comment (tylersticka) [inline src/base/_typography.scss]: Can we include these `@font-face` declaration in a separate `_fonts.scss` file so they won't clutter our typographic styles?

File: src/base/_typography.scss [hunk 2]
```diff
@@ -14,4 +14,65 @@ html {
...
+      format('woff2'),
+    url('../src/fonts/source-sans-pro-latin-extended/italic.woff')
+      format('woff');
+  font-style: italic;
+}
+@font-face {
+  font-family: 'Source Code Pro';
+  src: url('../src/fonts/source-code-pro-latin/400.woff2') format('woff2'),
+    url('../src/fonts/source-code-pro-latin/400.woff') format('woff');
+  font-weight: 400;
```
Comment (tylersticka) [inline src/base/_typography.scss]: Idea: Maybe we could write a Sass mixin to keep these declarations a tad more DRY?

**Comment by tylersticka** [comment]
variable-whee

This is so much fun!

**Comment by calebeby** [comment]
> variable-whee
> 
> This is so much fun!

That is so cool! I didn't know firefox had a slider for that

File: src/base/_fonts.scss [hunk 3]
```diff
@@ -0,0 +1,49 @@
+@mixin _non-variable-font-face($name, $path, $parameter) {
```
Comment (tylersticka) [inline src/base/_fonts.scss]: I'm curious if the leading underscores are meant to signify something?

Comment (calebeby) [reply src/base/_fonts.scss]: Yes, it is used to make the mixin not exported from that file

https://sass-lang.com/documentation/at-rules/use#private-members

**Pull request merged**